### PR TITLE
Bug/38 wait for step with step by step

### LINKED ISSUE #38: [Bug]: Default ActionID "wait for step" does not work for single motion
##PosTable version

v0.2.0

##Current behavior

If `nMode=2` (single motion) and an entry of PosTable has the default ActionID "wait for step" (`nActionID=2`), the `McePosTable` is stuck in state 31.
This happens because the step by step mode and the ActionID both needs a rising edge to continue.

##Expected behavior

A single rising edge on `bStep` should continue the trajectory.

### PR DESCRIPTION
**Pull request by yeu-rioual**
Fixes #38 

Also fixes a bug where a motion and its actionID run a second time if servos are disabled then enabled while the actionID was running.

### CODE REVIEW

File: source/examples/functions/McePosTable/definition.yaml [hunk 1]
```diff
@@ -1,6 +1,17 @@
 author: Rioual
 comment: Performed a trajectory stored in a PosTable
 changelog:
+  - version: 0.6.0
+    date: 2024-09-30
+    author: Rioual
+    fixed:
+      - State machine stuck if `nMode = 2` and `WAIT_FOR_STEP` actionID is active
+      - Motion and actionID are run again if servos are disabled then enabled
```
Comment (yeu-degroot) [inline source/examples/functions/McePosTable/definition.yaml:9]: Perhaps change this to "Motion and actionID no longer being re-executed after recovery of `bSystemReady`"

**Comment by yeu-degroot** [comment]
What about #45? Can we merge that one first?
After that update (merge) `bug/38-wait-for-step-with-step-by-step` from `main`.

**Comment by yeu-rioual** [comment]
> What about #45? Can we merge that one first? After that update (merge) `bug/38-wait-for-step-with-step-by-step` from `main`.

Yes, `bug/38-wait-for-step-with-step-by-step` is a branch from #45 so it already include the variable renaming.

File: source/examples/functions/McePosTable/definition.yaml (hunk 1 above)
Comment (yeu-rioual) [reply source/examples/functions/McePosTable/definition.yaml]: Done in [b465bf2](https://github.com/YaskawaEurope/mlx-examples/pull/46/commits/b465bf2885f74496eda076449596517d9658da62)

**Pull request closed without merging**